src/mqt/qecc/circuit_synthesis/encoding.py

```
@@ -354,5 +749,296 @@ def gottesman_encoding_circuit(tableau: StabilizerTableau | list[str]) -> tuple[
+        stacked = np.vstack((independent_checks, row))
+        new_rnk = mod2.rank(stacked)
+        if new_rnk > prev_rnk:
+            independent_checks = stacked
+            prev_rnk = new_rnk
+        if prev_rnk == rnk:
+            break
+    return CheckMatrix(independent_checks, pauli_type=checks.type)
+
+
+def cnot_encoding_circuit(
+    checks: CheckMatrix, logicals: CheckMatrix, balance_checks: bool = False, config: SynthesisConfig | None = None
```

Ruff (0.15.12)

[warning] Boolean-typed positional argument in function definition

(FBT001)